vignettes/vrd_vignette.Rmd

```diff
@@ -0,0 +1,259 @@
+
+Used to retrieve a data frame listing summary data of recall(s) involving an item between for the provided recall number(s). It has **three** inputs:
+
+- recall_number: A seven digit number of the item recall. Can be a list.
+- limit: An integer that specifies how recalls you want to return. If left empty defaults to 25.
+- api_key: A string of the api key needed to access the database. If left empty will error and not provide requested data frame.
+
+**Example:** If you wanted to search how many items were recalled from the start of the database to the manufacturing year of 2010 you could use the following.
+```{r recall_details, eval=FALSE}
+recall_details(1977044)
+#>   RECALL_NUMBER_NUM MANUFACTURER_RECALL_NO_TXT CATEGORY_ETXT CATEGORY_FTXT MODEL_NAME_NM MAKE_NAME_NM UNIT_AFFECTED_NBR SYSTEM_TYPE_ETXT SYSTEM_TYPE_FTXT
+#> 1           1977044                                      Car       Voiture         PINTO         FORD                28      Not Entered       Non Saisie
+#>   NOTIFICATION_TYPE_ETXT NOTIFICATION_TYPE_FTXT
+#> 1            Not Entered             Non Saisie
+#>                                                                                                                                               COMMENT_ETXT
+#> 1 VEHICLES WERE BUILT WITH INCORRECT NUTS ATTACHING THE REAR BUMPER LEFT ISOLATOR TO THE FRAME.  THESE NUTS COULD CONTACT AND PUNCTURE THE FUEL TANK. \r\n
+#>                COMMENT_FTXT DATE_YEAR_CD RECALL_DATE_DTE
+#> 1 Translation not available         1977      1977-08-09
+```
+
+
+### Example of How to Use the Data
+
+Lets say we want to look in the data base for the companies that did the most recalls with the items they manufactured in the year 2000.  First we would start with querying the API for all items recalled that were manufactured in the year 2000.
+
+```{r recall, eval = FALSE}
+recall_2000 <- recall_by_years(start_year = 2000, end_year = 2000, limit = 10000)
+```
+
+We are putting the limit sufficiently high to assume there weren't more than 10,000 recalls that year in Canada. We can count up the data for each company by putting the `Make name` column into a table. We can then just look at only the table values that are larger than 65.
+
+```{r count, eval = FALSE}
+counts <- table(recall_2000$`Manufacturer Name`)
+counts <- counts[counts>=65]
+```
+
+```{r hidden, echo=FALSE}
+COA <- rep("COACHMEN", 70)
+DAIM <- rep("DAIMLERCHRYSLER", 66)
+FORD <- rep("FORD", 151)
+FRE <- rep("FREIGHTLINER", 189)
+GM <- rep("GENERAL MOTORS", 239)
+INT <- rep("INTERNATIONAL", 170)
+
+counts <- table(c(COA, DAIM, FORD, FRE, GM, INT))
+```
+
+This reduces us to 11 companies. We could then plot this in a bar plot as seen below.
+
+```{r plot, fig.width=7, fig.height=5, fig.align='center'}
+barplot(counts,
+        main="Top Company Recalls in 2000",
+        xlab="Companies",
+        ylab= "Total Recalls",
+        cex.names = 0.5,
+        border = TRUE,
+        col = "lightblue")
+```
+
```

Comment:
I think this section needs a bit more beef and some good plots as well, probably cut down the functions stuff as described in one of the comments above